Update UG (#293)
AY2223S2-CS2103T-T15-4/tp: docs/UserGuide.md

2. Copy the file path of the **folder** you put `clipboard.jar` in.
    * [Guide For Windows](https://www.howtogeek.com/670447/how-to-copy-the-full-path-of-a-file-on-windows-10/)
    * [Guide For MacOS](https://support.apple.com/en-sg/guide/mac-help/mchlp1774/mac#:~:text=Show%20the%20path%20to%20a%20file%20or%20folder&text=Choose%20View%20%3E%20Show%20Path%20Bar,bottom%20of%20the%20Finder%20window.)
    * [Guide For Linux](https://www.tutorialspoint.com/get-the-full-path-of-a-file-in-linux)
3. In the command terminal, type `cd <FILE_PATH>`, where `<FILE_PATH>` is the file path you found in step 2.
4. Type `java -jar clipboard.jar` into the command terminal and press `Enter` to launch CLIpboard.&nbsp;

If you see something like this, you're good to go!<br>
<img src="./images/Ui.png" width="650" height="529"/>
<br>
*CLIpboard contains some sample data if there is no initial data, as is the case when you first launch it.*

### Quick Start

1. When you first launch CLIpboard, you will be brought to the `Course page`.

> 2. Type your command in the `command box` and press `Enter/Return` to execute it.

Author: added link to visual guide

    In this command,
    - The prefix `n/` in `n/john doe` tells CLIpboard that this parameter is the student's name.
    - The prefix `p/` in `p/12345678` tells CLIpboard that this parameter is the student's phone number.
- **Parameters can be input in any order**<br>
  - If the command format specifies the required parameters as `n/NAME p/PHONE_NUMBER`, an alternative format like `p/PHONE_NUMBER n/NAME` is also acceptable.

<div markdown="span" class="alert alert-info">
  :information_source: **Note:** It is important to enter commands exactly in their required formats in order for CLIpboard to understand them.<br>
</div>

---

## General Features

General features are features that may be used anywhere on CLIpboard, regardless of what page you're on.

### Showing the home page: `home`

Brings you to the home page, which is the `Course Page` of CLIpboard.

**Command Format:**

```
home
```

[Back to list of features](#commands)

### Going back to the previous page: `back`

Brings you back to the previous page you were at.

**Command Format:**

```
back
```

[Back to list of features](#features)

### Undoing the last command you did: `undo`
Because everyone makes mistakes.

> You may `undo` up to 5 previous commands.

Author: thanks, will do

<br>

### Adding a course: `add course`
If you're teaching a new course, you can use this command to add it to the `Course Page`.

**Command Format:**

```
add course <COURSE>
```

<div markdown="block" class="alert alert-info">:information_source: **Note:**<br>
  * Adding courses with the same course code is not possible.
  * Course codes can contain alphanumeric and special characters. e.g. "AY2223S2-CS2103T"
  * This command is case-sensitive. e.g. "CS2103T" and "cs2103t" are treated as different courses.
</div>

Examples:

- `add course CS2103T` will add a new course with module code `CS2103T`
- `add course CS2105` will add a new course with module code `CS2105`

<br>

[Back to list of features](#features)

### Deleting a course: `delete course`
If you're done teaching a course, you can remove it with this command.

**Command Format:**

```
delete course <INDEX>
```

Examples:

- `delete course 1` will delete the first course listed in your course list.

[Back to list of features](#features)

### Editing a course: `edit course`

Use this command to change the course code of an existing course.

**Command Format:**

```
edit course <INDEX> <NEW COURSE NAME>
```

Examples:

> - `edit course 1 CS2106` will rename your first course listed in your course list to "CS2106".

Reviewer: ```suggestion
- `edit course 1 CS2106` will rename your *first* course listed in your course list to "CS2106".
```
Just a minor one for consistency. 
Have other instances as well, not adding more comments to not clutter this up as it's easily searchable.

> - `edit course 1 CS2106` will rename your first course listed in your course list to "CS2106".

Author: > Just a minor one for consistency.
> Have other instances as well, not adding more comments to not clutter this up as it's easily searchable.

just to clarify,
ive intentionally tried to be consistent to not italicisie the "first" in all examples of command formats. the only times i want to italicise them is in the sections that are before "Command Format" because the reader hasnt been introduced to what index means, so i draw extra emphasis there to show that the number for the command means something.

> - `edit course 1 CS2106` will rename your first course listed in your course list to "CS2106".

Reviewer: Ah then that makes sense 👍, no change required.

---

## Managing Groups
In the following sections, you will find the features that are available to you on the `Group Page`.
A `Group Page` typically looks like this:

<img src="./images/UiGroupPage.png" width="650" height="529"/>
<br>

### Adding a group: `add group`

Use this command to a new group to a course you've selected.

**Command Format:**

```
add group <GROUP>
```

<div markdown="block" class="alert alert-info">:information_source: **Note:**<br>
  * Adding groups with the same group name is not possible.
  * Group names can contain alphanumeric and special characters. e.g. "AY2223S2-CS2103T"
  * This command is case-sensitive. e.g. "CS2103T" and "cs2103t" are treated as different courses.
</div>

Examples:

- `add group T15` will add a new group with group name `T15`
- `add group L08` will add a new group with group name `L08`

[Back to list of features](#features)

### Deleting a group: `delete group`

This removes a group from your selected course.

**Command Format:**

```
delete group <INDEX>
```

Examples:

- `delete group 1` will delete the first group listed in your group list.

[Back to list of features](#features)

### Editing a group: `edit group`

> Use this command to edit the name of an existing group.

Author: there is already a note under the add commands that duplicates are not allowed. I think under normal use case a user wouldnt be editing to add duplicates unless its an accident? if its an accident then the error message will handle it.

lmk what u think

> **Command Format:**

Author: fixed

</div>
<br>

Examples:

- `add student n/Tan Wei Xuan p/81776544 e/twx@example.com sid/A6788796F`
- `add student n/Amirul Bin Othman p/98884417 e/amirul@example.com sid/A1324356G`

[Back to list of features](#features)

### Deleting a student: `delete student`

This removes a student from a selected group.

**Command Format:**

```
delete student <INDEX>
```

Examples:

- `delete student 1` will delete your first student listed in your student list.

[Back to list of features](#features)

### Editing a student: `edit student`

If your student has changed their phone number, you may edit it, or any other student information with this command.

**Command Format:**

```
edit student <INDEX> [n/<NAME>] [p/<PHONE_NUMBER>] [e/<EMAIL>] [sid/<STUDENT_NUMBER>]
```
<div markdown="block" class="alert alert-info">:information_source: **Note:**<br>

  * At least one parameter `[n/<NAME>]`, `[p/<PHONE_NUMBER>]`, `[e/<EMAIL>]` or `[sid/<STUDENT_NUMBER>]` must be provided.
  * Parameters entered following `edit <INDEX>` will replace the original fields.

>   * Parameters not entered will not replace the original fields.

Author: removed those 2 lines. not sure if they are necessary to include anyway

* Keywords are separated by spaces.
* Keywords are case-insensitive.
</div>
<br>

Examples:

- `find student John` returns `John`.
- `find student Alex Yu` returns `Alex Yeoh`, `Bernice Yu`.
- `find student A123` returns students whose student ID contains `A123`, such as `A123` and `A1234567X`.

[Back to list of features](#features)

### Sorting list of students: `sort`

You can sort your students either by their names or their student ID.

**Command Format:**

```
sort <CATEGORY>
```
- `<CATEGORY>` is either `name` or `id`

Examples:

- `sort name` will sort your list of students alphabetically according to name.
- `sort id` will sort your list of students alphanumerically according to student ID.

<img src="./images/UiStudentSortName.png" width="650" height="529"/>
<br>
<img src="./images/UiStudentSortId.png" width="650" height="529"/>
<br>

[Back to list of features](#features)

### Display a student's information: `select`

This command allows you to select a student and display their full particulars.
The selected student will be shown like this:

<img src="./images/UiStudentView.png" width="650" height="529"/>

**Command Format:**

```
select <INDEX>
```

Examples:

> - `select 1` will select the first student in the student list and display his / her particulars on the view panel on the right.

Author: removed mention of "view panel", to just be  "select the first student in the student list and display his / her particulars." .  

good catch. i actually meant to remove all mentions of right pane etc. since its abit extra as the user just needs to know that they will see it. help to see if theres any more of these.

[Back to list of features](#features)

### Adding or deleting a remark: `remark`

You can use this command to add, edit or remove remarks from a student.

**Command Format for Adding/Editing a remark:**

```

> remark <INDEX> [<REMARK>]

Author: fixed

You can view a summary of a student's attendance like this:

<img src="./images/UiStudentAttendanceView.png" width="650" height="529"/>

**Command Format:**

```
attendance
```

<div markdown="block" class="alert alert-info">:information_source: **Note:** To edit a student's attendance for a particular session, you can navigate to their [Attendance Page](#attendance-page-commands).
</div>
<br>

[Back to list of features](#features)

---

## Managing Sessions

In the following sections, you will find the features that are available to you on the `Session Page`.
A `Session Page` typically looks like this:
<img src="./images/UiSessionPage.png" width="650" height="529"/>
<br>

### Adding a session: `add session`

This adds a new session to your selected group.

**Command Format:**

```
add session <SESSION>
```

<div markdown="block" class="alert alert-info">:information_source: **Note:**<br>

* Adding sessions with the same session name is not possible.
* Session names can contain alphanumeric and special characters. e.g. "T15-Session-1.2"
* This command is case-sensitive. e.g. "session1" and "SESSION1" are treated as different sessions.

> </div>

Author: added, and also added for course & group. (task name allows it)

mark <INDEX>...
```

Examples:

- `mark 1` will mark the first student as present.
- `mark 2,4,5,6` will mark the 2nd, 4th, 5th and 6th students as present.

[Back to list of features](#features)

### Marking students as absent: `unmark`

Marks selected students' attendance as absent for current session.

**Command Format:**

```
unmark <INDEX>
unmark <INDEX>...
```

Examples:

- `unmark 1` will mark your 1st student as absent.
- `unmark 2,3,4` will mark your 2nd, 3rd and 4th students as absent.

[Back to list of features](#features)

### View the overall attendance of a session: `attendance`

You can view the overall attendance of a session that you've selected, which will be shown like this:

<img src="./images/UiAttendance.png" width="650" height="529"/>
<br>

**Command Format:**

```
attendance
```

[Back to list of features](#features)

---

## Managing Tasks
In the following sections, you will find the features that are available to you on the `Task Page`.
A `Task Page` typically looks like this:
<img src="./images/UiTaskPage.png" width="650" height="529"/>
<br>

### Adding a task: `add task`

This adds a new task to your selected group.

**Command Format:**

```

> add task <TASK NAME>

Author: thanks. fixed and replaced all white spaces in parameters to be underscores

#### Launching CLIpboard
*Option 1*:
* Double-click on `clipboard.jar` to run it. If this does not work, go to *Option 2*.

*Option 2*:
1. Open a command terminal. [Guide to open a command terminal](https://www.ionos.com/help/email/troubleshooting-mail-basicmail-business/access-the-command-prompt-or-terminal/#:~:text=Open%20Command%20Prompt%20in%20Windows,cmd%22%20and%20then%20click%20OK.).
2. Copy the file path of the **folder** you put `clipboard.jar` in.
    * [Guide For Windows](https://www.howtogeek.com/670447/how-to-copy-the-full-path-of-a-file-on-windows-10/)
    * [Guide For MacOS](https://support.apple.com/en-sg/guide/mac-help/mchlp1774/mac#:~:text=Show%20the%20path%20to%20a%20file%20or%20folder&text=Choose%20View%20%3E%20Show%20Path%20Bar,bottom%20of%20the%20Finder%20window.)
    * [Guide For Linux](https://www.tutorialspoint.com/get-the-full-path-of-a-file-in-linux)
3. In the command terminal, type `cd <FILE_PATH>`, where `<FILE_PATH>` is the file path you found in step 2.
4. Type `java -jar clipboard.jar` into the command terminal and press `Enter` to launch CLIpboard.&nbsp;

If you see something like this, you're good to go!<br>

> <img src="./images/Ui.png" width="650" height="529"/>

Reviewer: This image looks better with the colours to show in our UG, but would it be better to show the `UiCoursePage.png` instead? Since it is what they would see when they first launch it.

Author: good catch. that was my intention but chose wrong file

<br>
*CLIpboard contains some sample data if there is no initial data, as is the case when you first launch it.*

### Quick Start

1. When you first launch CLIpboard, you will be brought to the `Course page`.
2. Type your command in the `command box` and press `Enter/Return` to execute it.
   <br>e.g. typing `help` and pressing the `Enter/Return` key will open the help window.
   <br>Some example commands you can try:
   1. `select 1` : Selects the first course displayed (eg. `CS2103T`) on the `Course Page`. You will be redirected to the `Group Page`.
   2. `select 1` : Selects the first group displayed (eg. `T15`) on the `Group Page`. You will be redirected to the `Student Page`.
   3. `add student n/John Doe p/98765432 e/johnd@example.com sid/A1234567X` : Adds a student named `John Doe` with the particulars into the list on the `Student Page`.

>    4. `delete student 3` : Deletes the 3rd student (eg. `Lim Kim Choo`) shown in the student list.

Reviewer: ```suggestion
   4. `delete student 3` : Deletes the *third* student (eg. `Lim Kim Choo`) shown in the student list.
```

For consistency with updated document.

Author: fixed, and replaced all other instances of numbered sequence to words

- [Managing Attendance: ](#attendance-page-commands)
  - [Marking attendance of a selected students: ](#marking-selected-students-as-present-mark) `mark`
  - [Unmarking attendance of a selected students: ](#marking-selected-students-as-absent-unmark) `unmark`
  - [View the overall attendance of a session: ](#view-the-overall-attendance-of-a-session-attendance) `attendance`
- [Managing Tasks: ](#managing-tasks)
  - [Adding a task: ](#adding-a-task-add-task) `add task`
  - [Deleting a task: ](#deleting-a-task-delete-task) `delete task`
  - [Editing a task: ](#editing-a-task-edit-task) `edit task`
  - [Selecting a task to assign grades: ](#selecting-a-task-to-assign-grades-select) `select`
  - [Finding a task: ](#finding-a-task-find-task) `find task`
- [Managing Grades: ](#managing-the-grades-of-a-task)
  - [Assigning a grade: ](#assigning-a-grade-to-a-student-assign) `assign`

### Commands

Features in CLIpboard are supported by commands.
Commands are like giving instructions to CLIpboard on what to do.

A typical command would look like this:
<br>
<img src="./images/typicalcommand.png" width="400" height="138"/>

> **Command keyword(s)**: Typically 1 or 2 words, they specify the type of action for CLIpboard to execute

Reviewer: ```suggestion
**Command keyword(s)**: Typically 1 or 2 words, they specify the type of action for CLIpboard to execute <br>
```

Author: fixed

**Supplementary command information**: These provide additional context about a command you wish to execute. Conventionally, these are referred to as "arguments" or "parameters".

Putting it all together for the above example, `find course` would tell CLIpboard that you're looking for a course, and "cs2101" tells CLIpboard which specific course you're looking for.

<div markdown="block" class="alert alert-warning">:exclamation: **Caution:** <br>

* If a command expects a parameter to be input only once, but multiple instances of it are included, only the last occurrence of the parameter will be taken.<br>
  * e.g. if you specify `p/12341234 p/56785678`, only `p/56785678` will be taken.
* Extraneous parameters for commands that do not expect any parameters will be ignored.<br>
  * e.g. if you input the command `help 123`, CLIpboard will interpret it simply as `help`.

> </div>

Reviewer: Would it be better for this to be a `Tip` instead of a `Caution`?

Author: i've defined tip to be specifically about improving a user's skills. And caution is defined as critical info abt clipboard behaviour. i think under this definition would fit better as caution

#### Understanding Command Formats
To help you understand how CLIpboard recognises commands, these are some of the conventions used in this guide when specifying command formats.

- **Supplementary command information**: These are represented in uppercase and enclosed by angle brackets (`<>`).<br>
  - e.g. in `add course <COURSE>`, `<COURSE>` is a parameter. `<COURSE>` can be replaced with a course name to form the command `add course CS2103T`.
- **Square brackets (`[]`)**: Parameters in square brackets are optional.<br>
  - e.g If the parameters are specified as `[n/<NAME>] [p/<PHONE_NUMBER>]`, omitting these parameters are acceptable.

> - **Elipses(`...`)**: Parameters postfixed by `…` can have multiple inputs.<br>
>   - e.g. `mark <INDEX>…` can be used as `mark 1` or `mark 1,2,3` etc.

Reviewer: ```suggestion
- **Elipses(`…`)**: Parameters postfixed by `…` can have multiple inputs.<br>
  - e.g. `mark <INDEX>…` can be used as `mark 1` or `mark 1,2,3` etc.
```

Just a minor one for consistency.

Author: thanks, fixed

**Command Format:**

```
home
```

[Back to list of features](#commands)

### Going back to the previous page: `back`

Brings you back to the previous page you were at.

**Command Format:**

```
back
```

[Back to list of features](#features)

### Undoing the last command you did: `undo`
Because everyone makes mistakes.

You may `undo` up to 5 previous commands.

**Command Format:**

```
undo
```

[Back to list of features](#features)

### Help: `help`
When you're feeling lost or unsure of how to do something, use this command to open the help window.
There, you can see a list of available commands on your current page, as well as a link to this User Guide.

**Command Format:**

```
help
```

[Back to list of features](#features)

### Clearing all entries: `clear`

Clears all entries in the list of the current page. Useful for when you've ended the curriculum and you wish to delete all data.

**Command Format:**

```
clear
```

<div markdown="span" class="alert alert-primary">:bulb: **Tip:**
Accidentally cleared your data? Don't worry, just run the <code>undo</code> command!
</div>

<div markdown="span" class="alert alert-warning">:exclamation: **Caution:**

> Be very careful with this command. Deleted data may not be recovered after closing the application or if you've gone beyond the `undo` limit.

Reviewer: ```suggestion
Be very careful with this command. Deleted data may not be recovered after closing the application or if you've gone beyond the `undo` limit of up to 5.
```

If user don't read this in sequence, they may not know what the `limit` in terms of number of times is, or can misinterpret what `limit` actually means.

Author: added

```
exit
```

[Back to list of features](#features)

---

## Managing Courses
In the following sections, you will find the features that are available to you on the `Course Page`.
A `Course Page` typically looks like this:

<img src="./images/UiCoursePage.png" width="650" height="529"/>
<br>

### Adding a course: `add course`
If you're teaching a new course, you can use this command to add it to the `Course Page`.

**Command Format:**

```
add course <COURSE>
```

<div markdown="block" class="alert alert-info">:information_source: **Note:**<br>
  * Adding courses with the same course code is not possible.
  * Course codes can contain alphanumeric and special characters. e.g. "AY2223S2-CS2103T"
  * This command is case-sensitive. e.g. "CS2103T" and "cs2103t" are treated as different courses.
</div>

Examples:

- `add course CS2103T` will add a new course with module code `CS2103T`
- `add course CS2105` will add a new course with module code `CS2105`

<br>

[Back to list of features](#features)

### Deleting a course: `delete course`
If you're done teaching a course, you can remove it with this command.

**Command Format:**

```
delete course <INDEX>
```

Examples:

> - `delete course 1` will delete the first course listed in your course list.

Reviewer: ```suggestion
- `delete course 1` will delete the *first* course listed in your course list.
```
Just a minor one for consistency.

Reviewer: No formatting error, should not be emphasised here.

* This command is case-insensitive. (e.g. `find course CS2103T` and `find course cs2103t` will give you the same result)
* Keywords are separated by spaces.
</div>
<br>

Example:
  If your course list contains 2 courses "CS2103T" and "CS2101":
  - `find course CS` will show "CS2103T", "CS2101".
  - `find course 03` will show "CS2103T".

[Back to list of features](#features)

### Selecting a course: `select`
This command allows you to select a course and navigates you to its corresponding `Group Page`.

**Command Format:**

```
select <INDEX>
```

Examples:

- `select 1` will display a list of groups from the first course.

[Back to list of features](#features)

---

## Managing Groups
In the following sections, you will find the features that are available to you on the `Group Page`.
A `Group Page` typically looks like this:

<img src="./images/UiGroupPage.png" width="650" height="529"/>
<br>

### Adding a group: `add group`

Use this command to a new group to a course you've selected.

**Command Format:**

```
add group <GROUP>
```

<div markdown="block" class="alert alert-info">:information_source: **Note:**<br>
  * Adding groups with the same group name is not possible.

>   * Group names can contain alphanumeric and special characters. e.g. "AY2223S2-CS2103T"

Reviewer: ```suggestion
  * Group names can contain alphanumeric and special characters. e.g. "L03-2"
```
Would be better to change the module to group names here.

Author: thanks fixed

>   * This command is case-sensitive. e.g. "CS2103T" and "cs2103t" are treated as different courses.

Reviewer: ```suggestion
  * This command is case-sensitive. e.g. "T15" and "t15" are treated as different courses.
```

Would be better to change the module to group names here.

  Adding a new student to a group will automatically add them to all existing sessions, but deleting a student from a group will NOT delete them from existing sessions (to preserve historical data).
</div>
<br>

Examples:

- `session 1` will show you your list of sessions from the first group.

[Back to list of features](#features)

### Displaying the Tasks of a Group: `task`
Use this command to view the tasks of a particular group.

**Command Format:**

```
task <INDEX>
```

<div markdown="span" class="alert alert-warning">:exclamation: **Caution:**
  Adding a new student to a group will automatically add them to all existing tasks, but deleting a student from a group will NOT delete them from existing tasks (to preserve historical data).
</div>
<br>

Examples:

- `task 1` will show you your list of tasks for the first group.

[Back to list of features](#features)

### Finding a group: `find group`
You can filter your group list to show only groups that contain specific keywords.

**Command Format:**

```
find group <KEYWORD> [<MORE_KEYWORDS>]
```

<div markdown="block" class="alert alert-info">:information_source: **Note:**<br>

> * This command is case-insensitive. (e.g. `find course T15` and `find course T15` will give you the same result)

Reviewer: ```suggestion
* This command is case-insensitive. (e.g. `find course T15` and `find course t15` will give you the same result)
```

Author: fixed

### Adding a student: `add student`

Use this command to add a new student to a selected group.

**Command Format:**

```
add student n/<NAME> p/<PHONE_NUMBER> e/<EMAIL> sid/<STUDENT_ID>
```

<div markdown="block" class="alert alert-info">:information_source: **Note:**<br>

  * Adding students with the same student id is not possible.
  * The same student can exist in multiple groups. For example, a student can be in both tutorial group "T1" and a project group "Team15" at the same time.
</div>
<br>

Examples:

- `add student n/Tan Wei Xuan p/81776544 e/twx@example.com sid/A6788796F`
- `add student n/Amirul Bin Othman p/98884417 e/amirul@example.com sid/A1324356G`

[Back to list of features](#features)

### Deleting a student: `delete student`

This removes a student from a selected group.

**Command Format:**

```
delete student <INDEX>
```

Examples:

- `delete student 1` will delete your first student listed in your student list.

[Back to list of features](#features)

### Editing a student: `edit student`

> If your student has changed their phone number, you may edit it, or any other student information with this command.

Reviewer: ```suggestion
If your student has changed their phone number or any other student information, you may edit it with this command.
```

Would it be better to arrange it like this?

Author: thanks , fixed

**Command Format:**

```
edit student <INDEX> [n/<NAME>] [p/<PHONE_NUMBER>] [e/<EMAIL>] [sid/<STUDENT_NUMBER>]
```
<div markdown="block" class="alert alert-info">:information_source: **Note:**<br>

  * At least one parameter `[n/<NAME>]`, `[p/<PHONE_NUMBER>]`, `[e/<EMAIL>]` or `[sid/<STUDENT_NUMBER>]` must be provided.
  * Parameters entered following `edit <INDEX>` will replace the original fields.
  * Parameters not entered will not replace the original fields.
</div>
<br>

> <div markdown="span" class="alert alert-primary">:bulb: **Tip:** Made a typo? Try the <code>undo</code> command!

Reviewer: The `<br>` should come after `Tip` instead.

Currently it looks like this:
<img width="760" alt="image" src="https://user-images.githubusercontent.com/71481877/231633570-e5ea3ea2-e471-4c59-a90e-f98bdb195468.png">

What it would be better to look like:
<img width="761" alt="image" src="https://user-images.githubusercontent.com/71481877/231633647-d8e9628f-78dc-4728-af43-4207aefce4d2.png">

Author: fixed

<br>
<img src="./images/UiStudentSortId.png" width="650" height="529"/>
<br>

[Back to list of features](#features)

### Display a student's information: `select`

This command allows you to select a student and display their full particulars.
The selected student will be shown like this:

<img src="./images/UiStudentView.png" width="650" height="529"/>

**Command Format:**

```
select <INDEX>
```

Examples:

- `select 1` will select the first student in the student list and display his / her particulars on the view panel on the right.

[Back to list of features](#features)

### Adding or deleting a remark: `remark`

You can use this command to add, edit or remove remarks from a student.

**Command Format for Adding/Editing a remark:**

```
remark <INDEX> [<REMARK>]
```

<div markdown="span" class="alert alert-warning">:exclamation: **Caution:** If there is already an existing remark, it will be replaced with the new remark.
</div>

**Command Format for deleting a remark:**

```
remark <INDEX>
```

<div markdown="span" class="alert alert-primary">:bulb: **Tip:**
  Accidentally deleted a remark? Don't worry, Try using the <code>undo</code> command!
</div>
<br>

Examples:

> - `remark 1 Loves watching Sci-Fi movies` will add a remark of 'Loves watching Sci-Fi movies' to the first student

Reviewer: ```suggestion
- `remark 1 Loves watching Sci-Fi movies` will add a remark of "Loves watching Sci-Fi movies" to the first student
```

Just a minor one for consistency with updated document.

Author: thanks, fixed

**Command Format:**

```
delete task <INDEX>
```

Examples:

- `delete task 1` will delete your first task listed.

[Back to list of features](#features)

### Editing a task: `edit task`

You can edit the name of an existing task in your task list.

**Command Format:**

```
edit task <INDEX> <NEW TASK NAME>
```

Examples:

- `edit task 1 CA5` will rename your first task listed to `CA5`.

[Back to list of features](#features)

### Finding a task: `find task`

You can filter your task list to show only tasks that contain specific keywords.

**Command Format:**

```
find task <KEYWORD> [<MORE_KEYWORDS>]
```

<div markdown="block" class="alert alert-info">:information_source: **Note:**<br>

* This command is case-insensitive. (e.g. `find task op1` and `find task OP1`
  will give you the same result)
* Keywords are separated by spaces.
</div>
<br>

Examples:

- `find task 1` will show `OP1` and `Critical Reflection 1`.

[Back to list of features](#features)

### Selecting a task to assign grades: `select`

This command allows you to select a task and view the grades of your students with that task.

**Command Format:**

```
select <INDEX>
```

> - Selects task at specified index and displays list of students with the task.

Reviewer: ```suggestion
- Selects task at specified index and displays list of students with the task. <br>
```

Examples:

- `select 1` will show a list of students who were assigned the first task on the list.

[Back to list of features](#features)

---

## Managing the Grades of a Task
In the following sections, you will find the features that are available to you after you have selected a task while on the `Task Page`.
Your current page would look similar to this:
<img src="./images/UiGradesPage.png" width="650" height="529"/>
<br>

<div markdown="span" class="alert alert-warning">:exclamation: **Caution:** The grades for all students will be set to ZERO when a task is created. Make sure to check that you have assigned grades to all students, so that no one's grade is accidentally left as 0.
</div>

### Assigning a grade to a student: `assign`

This assigns a grade to a student.

**Command Format:**

```
assign <INDEX> <GRADE>
```
<div markdown="span" class="alert alert-info">:information_source: **Note:** A grade must be a number between `0 and 100`.
</div>
<br>

Examples:

> - `assign 1 78` will assign the first student on the Student Page the grade of 78.

Reviewer: ```suggestion
- `assign 1 78` will assign the first student on the right pane the grade of 78.
```

Similarly for below.
